### PR TITLE
Fixed bugs after dnn refactoring. Added new tests for layers

### PR DESCRIPTION
**Pull request by arrybn**
### This pullrequest changes
Fixed some bugs after refactoring dnn. Added new tests, testdata is here: https://github.com/opencv/opencv_extra/pull/334

### CODE REVIEW

**Comment by alalek** [comment]
I have enabled DNN tests on CI. Could you take a look on these failures?

**Comment by arrybn** [comment]
Tests failed due to lack of torch test files - they were in another PR. But now I merged that PR with the current

**Comment by arrybn** [comment]
All tests have been passed. But size of data I want to add into opencv_extra is beyond the limits, especially because of many test data for torch

**Comment by mshabunin** [comment]
There are still several failures left on Windows:
```
[ RUN      ] Torch_Importer.net_cadd_table
unknown file: error: C++ exception with description "...\modules\dnn\src\torch\torch_importer.cpp:751: error: (-213) Unsupported Torch class "

V" in function cv::dnn::TorchImporter::readTorchObject
" thrown in the test body.
[  FAILED  ] Torch_Importer.net_cadd_table (8 ms)
```

**Comment by arrybn** [comment]
I'll fix them soon

**Comment by mshabunin** [comment]
@arrybn , looks like Git converts all text files to CR-LF format which is not supported by THDiskFile. Try to add `.gitattributes` file to the root of the _opencv_extra_ repository with the following content (https://git-scm.com/docs/gitattributes) to disable automatic conversion:
```
testdata/dnn/torch/*.txt text eol=lf
```

**Comment by arrybn** [comment]
Thank you

File: modules/dnn/src/torch/THFile.h [hunk 1]
```diff
@@ -28,15 +28,15 @@ TH_API unsigned char THFile_readByteScalar(THFile *self);
 TH_API char THFile_readCharScalar(THFile *self);
 TH_API short THFile_readShortScalar(THFile *self);
 TH_API int THFile_readIntScalar(THFile *self);
-TH_API long THFile_readLongScalar(THFile *self);
+TH_API int64 THFile_readLongScalar(THFile *self);
```
Comment (mshabunin) [inline modules/dnn/src/torch/THFile.h]: Looks like you have to `#include "opencv2/core/hal/interface.h"` to use `int64` type.

Comment (arrybn) [inline modules/dnn/src/torch/THFile.h]: Fixed, thanks

**Comment by mshabunin** [comment]
:+1:

**Pull request merged**